src/main/java/com/mercadopago/client/oauth/OauthClient.java

```diff
@@ -0,0 +1,161 @@
+package com.mercadopago.client.oauth;
+
+import com.mercadopago.MercadoPagoConfig;
+import com.mercadopago.client.IdempotentRequest;
+import com.mercadopago.client.MercadoPagoClient;
+import com.mercadopago.client.user.UserClient;
+import com.mercadopago.core.MPRequestOptions;
+import com.mercadopago.exceptions.MPException;
+import com.mercadopago.net.HttpMethod;
+import com.mercadopago.net.MPHttpClient;
+import com.mercadopago.net.MPResponse;
+import com.mercadopago.net.UrlFormatter;
+import com.mercadopago.resources.oauth.OauthCredential;
+import com.mercadopago.resources.user.User;
+import com.mercadopago.serialization.Serializer;
+import java.util.Objects;
+import java.util.Optional;
+
+/** Client responsible for performing oauth authorization. */
+public class OauthClient extends MercadoPagoClient {
+  private final UserClient userClient;
+
+  private final String authHost = "https://auth.mercadopago.com";
```

Comment:
Nesses casos, não seria melhor declarar o valor como uma constante? 
```suggestion
  private static final String AUTH_HOST = "https://auth.mercadopago.com";
```

Reply:
Essa classe não deve ser usada com frequência. Se a declaro estática, ela vai ser carregada pra todo mundo sem necessidade. Faz sentido?